### PR TITLE
E2E test for tokens only namespaces

### PR DESCRIPTION
**Pull request by shorsher**
* Adds an E2E test for namespaces with only a `tokens` plugin
* Moves `keynormalization` config under `namespaces.predefined[].assets.manager.keynormalization`
* Adds common for code adding namespaces, resetting firefly, etc. to a common package

### CODE REVIEW

File: internal/coreconfig/coreconfig.go [hunk 1]
```diff
@@ -41,6 +41,8 @@ const (
 	NamespacePlugins = "plugins"
 	// NamespaceDefaultKey is the default signing key for blockchain transactions within this namespace
 	NamespaceDefaultKey = "defaultkey"
+	// NamespaceAssetKeyNormalization mechanism to normalize keys before using them. Valid options: "blockchain_plugin" - use blockchain plugin (default), "none" - do not attempt normalization
+	NamespaceAssetKeyNormalization = "assets.manager.keynormalization"
```
Comment (awrichar) [inline internal/coreconfig/coreconfig.go:45]: The global one below is `asset` and not `assets` - probably should match?

File: test/e2e/gateway/tokens_only.go [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+func (suite *TokensOnlyTestSuite) SetupSuite() {
+	suite.testState = beforeE2ETest(suite.T())
+	stack := e2e.ReadStack(suite.T())
+	stackState := e2e.ReadStackState(suite.T())
+
+	adminProtocol1 := schemeHTTP
+	if stack.Members[0].UseHTTPS {
+		adminProtocol1 = schemeHTTPS
+	}
+	suite.adminHost1 = fmt.Sprintf("%s://%s:%d", adminProtocol1, stack.Members[0].FireflyHostname, stack.Members[0].ExposedAdminPort)
```
Comment (awrichar) [inline test/e2e/gateway/tokens_only.go:51]: A lot of this stuff is now in `testState` - see https://github.com/hyperledger/firefly/commit/6a2fc3cff8ac31d3810c73e28bbd0eb0d75037dc

Comment (shorsher) [reply test/e2e/gateway/tokens_only.go:51]: I think it's only in `multiparty` `beforeE2E()`, but I can do the same here

**Comment by shorsher** [comment]
Adding the `migration_consideration` label because `asset.manager.keynormalization` config has been moved under `namespaces.predefined[].asset.manager.keynormalization`

File: docs/reference/config.md [hunk 3]
```diff
@@ -30,7 +30,7 @@ nav_order: 2
 
 |Key|Description|Type|Default Value|
 |---|-----------|----|-------------|
-|keyNormalization|Mechanism to normalize keys before using them. Valid options are `blockchain_plugin` - use blockchain plugin (default) or `none` - do not attempt normalization|`string`|`<nil>`
+|keynormalization|Mechanism to normalize keys before using them. Valid options are `blockchain_plugin` - use blockchain plugin (default) or `none` - do not attempt normalization (deprecated - namespace.predefined[].asset.manager.keynormalization)|`string`|`<nil>`
```
Comment (awrichar) [inline docs/reference/config.md:33]: `namespaces` not `namespace`
and for clarity might want to say "_use_ namespaces.predefined[].asset.manager.keynormalization"

File: internal/identity/identitymanager.go [hunk 4]
```diff
@@ -135,7 +135,7 @@ func (im *identityManager) NormalizeSigningKey(ctx context.Context, inputKey str
 	}
 	// If the caller is not confident that the blockchain plugin/connector should be used to resolve,
 	// for example it might be a different blockchain (Eth vs Fabric etc.), or it has its own
-	// verification/management of keys, it should set `assets.keyNormalization: "none"` in the config.
+	// verification/management of keys, it should set `namespace.predefined[].asset.manager.keynormalization: "none"` in the config.
```
Comment (awrichar) [inline internal/identity/identitymanager.go:138]: `namespaces`

File: test/e2e/gateway/common.go [hunk 5]
```diff
@@ -41,9 +40,16 @@ type testState struct {
 	startTime            time.Time
 	done                 func()
 	ws1                  *websocket.Conn
+	ws2                  *websocket.Conn
 	client1              *client.FireFlyClient
+	client2              *client.FireFlyClient
```
Comment (awrichar) [inline test/e2e/gateway/common.go:45]: We never have 2 clients in a gateway test, right?

File: test/e2e/gateway/tokens_only.go [hunk 6]
```diff
@@ -0,0 +1,128 @@
...
+
+func (suite *TokensOnlyTestSuite) SetupSuite() {
+	suite.testState = beforeE2ETest(suite.T())
+	stack := e2e.ReadStack(suite.T())
+	stackState := e2e.ReadStackState(suite.T())
+
+	suite.connector = stack.TokenProviders[0]
+	suite.db = stack.Database
+	account := stackState.Accounts[0].(map[string]interface{})
+	suite.key = account["address"].(string)
```
Comment (awrichar) [inline test/e2e/gateway/tokens_only.go:46]: Do we need to extract this rather than just using the default key?

**Review by awrichar — APPROVED** [review]
Left a few more questions/comments, but putting an approval on this to merge after you go through them.

File: test/e2e/gateway/tokens_only.go (hunk 6 above)
Comment (shorsher) [reply test/e2e/gateway/tokens_only.go:46]: Looking through the code, is there any better way to get the `defaultkey`? I don't see it used anywhere.

Comment (awrichar) [reply test/e2e/gateway/tokens_only.go:46]: You should be able to just leave `key` blank and asset manager will always use the default.

**Pull request merged**